examples/observation_collection_for_imitation_learning.py

```diff
@@ -49,40 +49,42 @@ def main(scenarios: Sequence[str], headless: bool, seed: int):
         traffic_sim=SumoTrafficSimulation(headless=headless, auto_start=True),
         envision=None if headless else Envision(),
     )
-    scenarios_iterator = Scenario.scenario_variations(
-        scenarios,
-        list([]),
+
+    total_scenario_variations = Scenario.discover_traffic_history_scenarios_count(
+        scenarios
     )
 
-    scenario = next(scenarios_iterator)
-    obs = smarts.reset(scenario)
+    scenarios_iterator = Scenario.scenario_variations(scenarios, [])
+    for _ in range(total_scenario_variations):
+        scenario = next(scenarios_iterator)
```

Comment:
Again same can be done here by consuming the iterator with `for`.
```suggestion
    for scenario in scenarios_iterator:
```